elixir/apps/domain/test/domain/resources_test.exs

```diff
@@ -156,10 +157,73 @@ defmodule Domain.ResourcesTest do
                address: ["can't be blank"],
                name: ["should be at most 255 character(s)"],
                filters: ["is invalid"],
-               connections: ["is invalid"]
+               connections: ["is invalid"],
+               type: ["can't be blank"]
              }
     end
 
+    test "validates dns address", %{subject: subject} do
+      attrs = %{"address" => String.duplicate("a", 256), "type" => "dns"}
+      assert {:error, changeset} = create_resource(attrs, subject)
+      assert "should be at most 253 character(s)" in errors_on(changeset).address
+
+      attrs = %{"address" => "a", "type" => "dns"}
+      assert {:error, changeset} = create_resource(attrs, subject)
+      refute Map.has_key?(errors_on(changeset), :address)
+    end
+
+    test "validates cidr address", %{subject: subject} do
+      attrs = %{"address" => "192.168.1.256/28", "type" => "cidr"}
+      assert {:error, changeset} = create_resource(attrs, subject)
+      assert "is not a valid CIDR range" in errors_on(changeset).address
+
+      attrs = %{"address" => "192.168.1.1", "type" => "cidr"}
```

Comment:
Minor note here: this is shorthand for `192.168.1.1/32` which we would want to allow

Reply:
@jamilbk I preferred to stay away from this shorthand syntax, it feels like a bad UI decision, if a user wants /32 (s)he should be explicit about that, otherwise for some people it will be expected while for others - weird behaviour.

Reply:
I see where you're coming from, but making the user add a /32 to their single IP resource will feel very weird indeed. Maybe we call this `inet` instead?

WireGuard and current Firezone allow single IPs to be entered without the /32, and then just normalize it to /32 for display.

Reply:
Not critical to add now though. We could also just append /32 in the UI before it hits the changeset to keep the backend simpler.